### ISSUE TITLE
Bug in the UI

### ISSUE THREAD

**Issue by philipstarkey** [labscript-suite-oldfinal1/runviewer#11]
**Original report ([archived issue](https://labscript-suite.github.io/labscript-suite-bitbucket-archive/#!/labscript_suite/runviewer/issues/11)) by Jan Werkmann (Bitbucket: [PhyNerd](https://bitbucket.org/%7B44c4905c-2b90-4045-a5f1-652b8e228626%7D/), GitHub: [PhyNerd](https://github.com/PhyNerd)).**

----------------------------------------
When canceling the select file dialog I get this error:
![Bildschirmfoto 2017-06-22 um 20.36.16.png](https://bitbucket-archive.labscriptsuite.org/data/bitbucket.org/repo/AeyynE/images/81279156-Bildschirmfoto_2017-06-22_um_20.36.16.png)

**Comment by philipstarkey** [comment]
**Original comment by Jan Werkmann (Bitbucket: [PhyNerd](https://bitbucket.org/%7B44c4905c-2b90-4045-a5f1-652b8e228626%7D/), GitHub: [PhyNerd](https://github.com/PhyNerd)).**

----------------------------------------
The second Bug should be fixed by replacing:

```
#!python
        self.last_opened_shots_folder = os.path.dirname(selected_files[0])
```

with :

```
#!python
        if len(selected_files)>0:
            self.last_opened_shots_folder = os.path.dirname(selected_files[0])
```

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
Are you sure about the first one? It looks like that variable is defined two lines earlier, and I'm not seeing any way it could be deleted in the intervening line:


```
#!python

                icon = QIcon(pixmap)
                colour_item.setData(lambda clist=self.shot_colour_delegate._colours, colour=colour: int_to_enum(clist, colour), Qt.UserRole)
                colour_item.setData(icon, Qt.DecorationRole)
```

**Comment by philipstarkey** [comment]
**Original comment by Jan Werkmann (Bitbucket: [PhyNerd](https://bitbucket.org/%7B44c4905c-2b90-4045-a5f1-652b8e228626%7D/), GitHub: [PhyNerd](https://github.com/PhyNerd)).**

----------------------------------------

* Edited issue description* changed title from "Bugs in the UI" to "Bug in the UI"

**Comment by philipstarkey** [comment]
**Original comment by Jan Werkmann (Bitbucket: [PhyNerd](https://bitbucket.org/%7B44c4905c-2b90-4045-a5f1-652b8e228626%7D/), GitHub: [PhyNerd](https://github.com/PhyNerd)).**

----------------------------------------
You are right, my bad. I missed deleting the lines beneath the else branch when manually updating my runviewer.

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
No problem, thanks for reporting the other one.

**Comment by philipstarkey** [comment]
**Original comment by Philip Starkey (Bitbucket: [pstarkey](https://bitbucket.org/%7B48af65db-e5fc-459c-a7eb-52eb1f9a5690%7D/), GitHub: [philipstarkey](https://github.com/philipstarkey)).**

----------------------------------------

* changed state from "new" to "resolved"

Merged in monashkrb/runviewer/bugfix ([pull request #10](https://bitbucket-archive.labscriptsuite.org/#!/labscript_suite/runviewer/pull-requests/10))

Fixed issue #11 where not selecting a file would result in an error

Approved-by: Philip Starkey <philip.starkey@monash.edu>

→ \<\<cset 056df5d0e2330145df2829c46889fc9ab0f3cbbd\>\>

**Comment by philipstarkey** [comment]
**Original comment by Philip Starkey (Bitbucket: [pstarkey](https://bitbucket.org/%7B48af65db-e5fc-459c-a7eb-52eb1f9a5690%7D/), GitHub: [philipstarkey](https://github.com/philipstarkey)).**

----------------------------------------
Fixed issue #11 where not selecting a file would result in an error

→ \<\<cset 740dc8a7624b2e2cc1c3f786c83313a8219e6c48\>\>

**Comment by philipstarkey** [comment]
**Original comment by Philip Starkey (Bitbucket: [pstarkey](https://bitbucket.org/%7B48af65db-e5fc-459c-a7eb-52eb1f9a5690%7D/), GitHub: [philipstarkey](https://github.com/philipstarkey)).**

----------------------------------------
Merged in monashkrb/runviewer/bugfix ([pull request #10](https://bitbucket-archive.labscriptsuite.org/#!/labscript_suite/runviewer/pull-requests/10))

Fixed issue #11 where not selecting a file would result in an error

Approved-by: Philip Starkey <philip.starkey@monash.edu>

→ \<\<cset 056df5d0e2330145df2829c46889fc9ab0f3cbbd\>\>
